crates/vm/levm/src/vm.rs

```diff
@@ -255,6 +256,64 @@ impl VM {
             self.env.refunded_gas,
         );
 
+        if is_precompile(&current_call_frame.code_address) {
+            let precompile_result = execute_precompile(current_call_frame);
+
+            match precompile_result {
```

Comment:
With the changes that I suggested now this would be a `Result<Bytes, VMError>`
We now have 3 different scenarios.
1. Ok(Bytes) => Return a transaction report with success result and the corresponding output.
2. Err(error)
  a. If error.is_internal() then we have to propagate it to the upper layer, we don't want to handle that
  b. Otherwise, we return a transaction report with revert reason, setting `gas_used = gas_limit`

Reply:
Done in [Change the function result handling in execute](https://github.com/lambdaclass/ethrex/pull/1454/commits/03cd9231bffb7fdf84bbfa1bd3458a47d8e2597d)